lib/tests/streamlit/runtime/metrics_util_test.py

```diff
@@ -211,3 +210,157 @@ def test_function(param1: int, param2: str, param3: float = 0.1) -> str:
         self.assertEqual(len(ctx.tracked_commands), 0)
         test_function(param1=10, param2="foobar")
         self.assertEqual(len(ctx.tracked_commands), 0)
+
+    @parameterized.expand(
+        [
+            (st._transparent_write, "magic"),
+            (st.experimental_memo.clear, "clear_memo"),
+            (st.experimental_singleton.clear, "clear_singleton"),
+            (st.session_state.__setattr__, "session_state.set_attr"),
+            (st.session_state.__setitem__, "session_state.set_item"),
+            (websocket_headers._get_websocket_headers, "_get_websocket_headers"),
+            (memo_decorator.MemoCache.write_result, "_cache_memo_object"),
+            (
+                singleton_decorator.SingletonCache.write_result,
+                "_cache_singleton_object",
+            ),
+            (caching._write_to_cache, "_cache_object"),
+        ]
+    )
+    def test_internal_api_commands(self, command: Callable, expected_name: str):
+        """Some internal functions are also tracked and should use the correct name."""
+        ctx = get_script_run_ctx()
+        ctx.reset()
+        ctx.gather_usage_stats = True
+
+        # This will always throw an exception because of missing arguments
+        # This is fine since the command still get tracked
+        with contextlib.suppress(Exception):
```

Comment:
Can we narrow the suppressed Exception type at all here? (So that if the command starts failing because of some _unexpected_ reason, our test fails?)

Reply:
Potentially it could be a lot of different exceptions depending on the executed callable which makes it a bit complicated to narrow it down. We are executing a callable without providing any parameters, but we don't really care that the wrapped function is actually able to execute successfully. We just care about the command being tracked which we check down below. So, from the perspective of the test, if there is anything unexpected (e.g. no command is tracked) the asserts below will alert us.